### ISSUE TITLE
Ability to force type & indexing analyzer for non-searchable fields

### ISSUE THREAD

**Issue by scottsb** [Smile-SA/elasticsuite#2085]
Currently it is possible to set the search analyzer in `elasticsuite_indices.xml` with the `defaultSearchAnalyzer` key, but this gets ignored if the field is not used for search or sorting:

https://github.com/Smile-SA/elasticsuite/blob/2.10.x/src/module-elasticsuite-core/Index/Mapping/Field.php#L328

Furthermore, it has the effect of also overriding our field type specification (`text` gets turned to `keyword`).

Our use case is a whitespace separated list of tokens (stored as a product attribute) that we have to filter all searches by (we have separate custom filters to do this). These tokens should not actually be directly searchable or sortable. Right now because the field is forced into a `keyword` field, the field is not analyzed, and a match to a single token does not work. Our current planned workaround is to add a plugin to `getMappingPropertyConfig()` that overrides the return value when it's our field, but that's not as ideal as if we could do it declaratively.

**Comment by romainruaud** [comment]
Hi @scottsb 

that's an interesting topic and I think it could improve Elasticsuite.

Since I'm answering with a lot of delay, maybe you implemented by yourself with a plugin in the meantime. 

Could you share what you ended up doing ?

Regards

**Comment by scottsb** [comment]
Yes, we wrote a plugin on `Smile\ElasticsuiteCore\Index\Mapping\Field`. Here's our plugin class, sanitized of our specific proprietary use case:

```
<?php
namespace Ripen\Example\Plugin;

use Smile\ElasticsuiteCore\Index\Mapping\Field;
use Ripen\Example\Model\Product;

class ForceESFieldProps
{
    /**
     * Workaround for this issue that prevents this from being defined in XML:
     * https://github.com/Smile-SA/elasticsuite/issues/2085
     *
     * @param Field $field
     * @param array $propertyConfig
     * @return array
     */
    public function afterGetMappingPropertyConfig(
        Field $field,
        array $propertyConfig
    ) {
        if (in_array($field->getName(), Product::EXAMPLE_FIELDS)) {
            $propertyConfig['type'] = Field::FIELD_TYPE_TEXT;
            $propertyConfig['analyzer'] = Field::ANALYZER_WHITESPACE;
            unset($propertyConfig['ignore_above']);
        }

        return $propertyConfig;
    }
}
```
